compiler/damlc/daml-prim-src/DA/Internal/Error.daml

```diff
@@ -0,0 +1,30 @@
+-- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+#ifndef DAML_EXCEPTIONS
+
+-- | HIDE
+module DA.Internal.Error where
+
+import GHC.Types ()
+
+#else
+
+-- | MOVE DA.Exception
+module DA.Internal.Error where
+
+import GHC.Types (Text)
+
+-- | Exception raised by `error`.
+data GeneralError = GeneralError { message : Text }
+
+-- | Exception raised by an arithmetic operation, such as divide-by-zero or overflow.
+data ArithmeticError = ArithmeticError { message : Text }
+
+-- | Exception raised when a contract is invalid, i.e. fails the ensure clause.
+data ContractError = ContractError { message : Text }
```

Comment:
maybe worth adding a todo to remove the message fields once we drop it in Daml-LF.

Reply:
Actually you already implement it based on the `Text`-based stuff so I don’t understand why you need this.

Reply:
I don't think we need to drop this field. We definitely want them for GeneralError, and it's simpler to have all these built-in execptions be the same than to have them be different. But we can also, I think, still support text messages in built-in exceptions, without having built-in exception types, since these can all carried by AnyException anyway. I think Rémy had said something like this at some point ... (Maybe we can talk about it next meeting.)